packages/contractkit/src/wrappers/BondedDeposits.ts

```diff
@@ -0,0 +1,127 @@
+import { zip } from '@celo/utils/lib/src/collections'
+import BN from 'bn.js'
+import { Address } from 'src/base'
+import { BondedDeposits } from 'src/generated/types/BondedDeposits'
+import { BaseWrapper } from 'src/wrappers/BaseWrapper'
+import Web3 from 'web3'
+import { TransactionObject } from 'web3/eth/types'
+
+export interface VotingDetails {
+  accountAddress: Address
+  voterAddress: Address
+  weight: BN
+}
+
+interface Deposit {
+  time: BN
+  value: BN
+}
+
+export interface Deposits {
+  bonded: Deposit[]
+  notified: Deposit[]
+  total: {
+    gold: BN
+    weight: BN
+  }
+}
+
+export class BondedDepositsWrapper extends BaseWrapper<BondedDeposits> {
+  async getAccountWeight(account: Address): Promise<BN> {
+    const accountWeight = await this.contract.methods.getAccountWeight(account).call()
+    return Web3.utils.toBN(accountWeight)
+  }
+
+  async getVotingDetails(_accountOrVoterAddress: Address): Promise<VotingDetails> {
+    throw new Error('Requires FIX on Incompatible Contract')
+  }
+  // FIXME this.contract.methods.getAccountFromVoter does not exist
+  // async getVotingDetails(accountOrVoterAddress: Address): Promise<VotingDetails> {
+  //   const accountAddress = await this.contract.methods
+  //     .getAccountFromVoter(accountOrVoterAddress)
+  //     .call()
+
+  //   return {
+  //     accountAddress,
+  //     voterAddress: accountOrVoterAddress,
+  //     weight: await this.getAccountWeight(accountAddress),
+  //   }
+  // }
+
+  async getBondedDepositValue(account: string, noticePeriod: string) {
+    const deposit = await this.contract.methods.getBondedDeposit(account, noticePeriod).call()
+    return this.getValueFromDeposit(deposit)
+  }
+
+  async zipAccountTimesAndValuesToDeposits(
```

Comment:
What do you think about making some of these functions private?

Reply:
sure... this was just a port of what is already present in cli